crates/zome_types/src/query.rs

```diff
@@ -21,6 +26,87 @@ pub struct ChainQueryFilter {
     pub include_entries: bool,
 }
 
+#[derive(Clone, Debug, PartialEq, serde::Serialize, serde::Deserialize, SerializedBytes)]
+/// An agents chain elements returned from a agent_activity_query
+pub struct AgentActivity {
+    /// Headers on this chain.
+    pub activity: Vec<Activity>,
+    /// The status of this chain.
+    pub status: ChainStatus,
+    /// The highest chain header that has
+    /// been observed by this authority.
+    pub highest_observed: Option<HighestObserved>,
+}
+
+#[derive(Clone, Debug, PartialEq, serde::Serialize, serde::Deserialize)]
+/// Individual agent activity.
+/// The header with it's validation status.
+// TODO: This is a little weird because when we have warrants
+// we will have a warrant for each position in the chain that
+// is warranted. Maybe that makes this redundant?
+pub struct Activity {
+    /// The header on this chain
+    pub header: SignedHeaderHashed,
+    /// The validation status of this individual header
+    pub validation_status: ValidationStatus,
+}
+
+#[derive(Clone, Debug, PartialEq, serde::Serialize, serde::Deserialize)]
+/// The highest header sequence observed by this authority.
+/// This also includes the headers at this sequence.
+/// If there is more then one then there is a fork.
+///
+/// This type is to prevent headers being hidden by
+/// withholding the previous header.
+///
+/// The information is tracked at the edge of holochain before
+/// validation (but after drop checks).
+pub struct HighestObserved {
+    /// The highest sequence number observed.
+    pub header_seq: u32,
+    /// Hashes of any headers claiming to be at this
+    /// header sequence.
+    pub hash: Vec<HeaderHash>,
+}
+
+#[derive(Clone, Debug, PartialEq, serde::Serialize, serde::Deserialize)]
+/// Status of the agent activity chain
+// TODO: In the future we will most likely be replaced
+// by warrants instead of Forked / Invalid so we can provide
+// evidence of why the chain has a status.
+pub enum ChainStatus {
+    /// This authority has no information on the chain.
+    Empty,
+    /// The chain is valid as at this header sequence and header hash.
+    Valid(ChainHead),
```

Comment:
Nitpicky and unimportant, but is there a reason why you want to create a new data type to pare this down to the absolute minimum, rather than just having a HeaderHashed?

Reply:
See the hackmd but the whole point of `ChainHead` is to be lightweight so you can call multiple authorities and see if they have the same answer. The whole header is really not needed (and would just get in the way) for this purpose.